services/wallet/async/async.go

```diff
@@ -155,3 +160,57 @@ func (d *AtomicGroup) Error() error {
 func (d *AtomicGroup) Stop() {
 	d.cancel()
 }
+
+func NewQueuedAtomicGroup(parent context.Context, limit int64) *QueuedAtomicGroup {
+	return &QueuedAtomicGroup{NewAtomicGroup(parent), limit, 0, []Command{}}
+}
+
+type QueuedAtomicGroup struct {
```

Comment:
I see some duplicate code between `AtomicGroup` and `QueuedAtomicGroup`.
I think some code in `AtomicGroup` can be split into functions that can then be called by `QueuedAtomicGroup` to avoid this. For example, it should be possible to let `AtomicGroup` handle the `WaitGroup` entirely instead of accessing it directly from `QueuedAtomicGroup`.